spinn_front_end_common/interface/interface_functions/chip_provenance_updater.py

```diff
@@ -59,15 +59,17 @@ def _run(self):
             left_to_do_cores,
             "Forcing error cores to generate provenance data")
 
-        cpu_infos = self.__txrx.get_cpu_infos(
+        bad_cores = self.__txrx.get_cores_in_states(
             self.__all_cores,
-            [CPUState.RUN_TIME_EXCEPTION, CPUState.WATCHDOG, CPUState.IDLE],
-            True)
-        error_cores = cpu_infos.infos_for_state(CPUState.RUN_TIME_EXCEPTION)
-        watchdog_cores = cpu_infos.infos_for_state(CPUState.WATCHDOG)
-        idle_cores = cpu_infos.infos_for_state(CPUState.IDLE)
-
-        if error_cores or watchdog_cores or idle_cores:
+            [CPUState.RUN_TIME_EXCEPTION, CPUState.WATCHDOG, CPUState.IDLE])
+
+        if bad_cores:
+            error_cores = self.__txrx.get_cores_in_states(
+                self.__all_cores, CPUState.RUN_TIME_EXCEPTION)
+            watchdog_cores = self.__txrx.get_cores_in_states(
+                self.__all_cores, CPUState.WATCHDOG)
+            idle_cores = self.__txrx.get_cores_in_states(
+                self.__all_cores, CPUState.IDLE)
```

Comment:
Similarly to a previous comment, get_cores_in_states returns the same information as get_cpu_infos so can't you still do the same thing rather than get them all again?